pkg/gljmain/gljmain.go

```diff
@@ -16,18 +16,136 @@ import (
 	"github.com/glojurelang/glojure/pkg/runtime"
 )
 
+// Args represents the parsed command line arguments
+type Args struct {
+	Mode         string   // "repl", "version", "help", "eval", "file"
+	Expression   string   // for eval mode
+	Filename     string   // for file mode
+	IncludePaths []string // for -I flags
+	CommandArgs  []string // remaining arguments after parsing
+}
+
+// parseArgs parses command line arguments and returns an Args struct
+func parseArgs(args []string) (*Args, error) {
+	if len(args) == 0 {
+		return &Args{Mode: "repl"}, nil
+	}
+
+	// First pass: collect all -I flags and their paths
+	var includePaths []string
+	var remainingArgs []string
+	var mode string
+	var expression string
+	var filename string
+
+	i := 0
+	for i < len(args) {
+		switch args[i] {
+		case "--version":
+			if mode == "" {
+				mode = "version"
+			}
+			i++
+		case "--help", "-h":
+			if mode == "" {
+				mode = "help"
+			}
+			i++
+		case "-e":
+			if mode == "" {
+				mode = "eval"
+				if i+1 >= len(args) {
+					return nil, fmt.Errorf("glj: -e requires an expression")
+				}
+				expression = args[i+1]
+				i += 2
+			} else {
+				remainingArgs = append(remainingArgs, args[i])
+				i++
+			}
+		case "-I":
+			if i+1 >= len(args) {
+				return nil, fmt.Errorf("glj: -I requires a path")
+			}
+			includePaths = append(includePaths, args[i+1])
+			i += 2
+		default:
+			if mode == "" {
+				mode = "file"
+				filename = args[i]
+			} else {
+				remainingArgs = append(remainingArgs, args[i])
+			}
+			i++
+		}
+	}
+
+	// If no explicit mode was set, default to repl
+	if mode == "" {
+		mode = "repl"
+	}
+
+	return &Args{
+		Mode:         mode,
+		Expression:   expression,
+		Filename:     filename,
+		IncludePaths: includePaths,
+		CommandArgs:  remainingArgs,
+	}, nil
+}
+
+// setupLoadPaths configures the library search path in order of priority
+func setupLoadPaths(includePaths []string) {
+	// Add GLJPATH directories to front of load path if set
+	loadPaths := os.Getenv("GLJPATH")
+	if loadPaths != "" {
+		paths := strings.Split(loadPaths, ":")
+		for i := len(paths) - 1; i >= 0; i-- {
+			path := paths[i]
+			if path != "" {
+				// Skip non-existent path directories
+				if _, err := os.Stat(path); err == nil {
+					runtime.AddLoadPath(os.DirFS(path), true)
+				}
+			}
+		}
```

Comment:
could we could drop the `true` second arg and start iterating from index 0 to get the same result?

Reply:
This all makes sense. I'll update in the morning.